editor/src/dev/dev.clj

```diff
@@ -754,3 +773,266 @@
+  []
+  (util/group-into
+    (sorted-map) (sorted-map)
+    (fn key-fn [[[_context-id cache-id] _entry-count]]
+      cache-id)
+    (fn value-fn [[[context-id _cache-id] entry-count]]
+      (pair context-id entry-count))
+    (scene-cache/cache-stats)))
+
+(set! *warn-on-reflection* false)
+
+(defn- buf-clj-attribute-data [^ByteBuffer buf ^long buf-vertex-attribute-offset ^long attribute-byte-size component-data-type]
+  (let [primitive-type-kw (buffers/primitive-type-kw component-data-type)
+        read-buf (-> buf
+                     (.slice buf-vertex-attribute-offset attribute-byte-size)
+                     (.order (.order buf))
+                     (buffers/as-typed-buffer component-data-type))]
+    (loop [clj-vector (vector-of primitive-type-kw)]
+      (if (.hasRemaining read-buf)
+        (let [attribute-component (.get read-buf) ; Return type differs by Buffer subclass.
+              clj-vector (conj clj-vector attribute-component)]
+          (recur clj-vector))
+        clj-vector))))
+
+(set! *warn-on-reflection* true)
+
+(defn- buf-clj-vertex-data [^ByteBuffer buf vertex-attributes ^long vertex-stride]
+  (let [buf-size (.limit buf)]
+    (assert (zero? (rem buf-size vertex-stride)) "Buffer size does not confirm to vertex stride.")
+    (mapv (fn [^long buf-vertex-offset]
+            (persistent!
+              (val
+                (reduce (fn [[^long buf-vertex-attribute-offset clj-vertex] vertex-attribute]
+                          (let [attribute-key (:name-key vertex-attribute)
+                                attribute-byte-size (vtx/attribute-size vertex-attribute)
+                                component-data-type (:type vertex-attribute)
+                                clj-vertex-attribute-data (buf-clj-attribute-data buf buf-vertex-attribute-offset attribute-byte-size component-data-type)
+                                clj-vertex-attribute (pair attribute-key clj-vertex-attribute-data)
+                                clj-vertex (conj! clj-vertex clj-vertex-attribute)
+                                buf-vertex-attribute-offset (+ buf-vertex-attribute-offset attribute-byte-size)]
+                            (pair buf-vertex-attribute-offset clj-vertex)))
+                        (pair buf-vertex-offset (transient []))
+                        vertex-attributes))))
+          (range 0 buf-size vertex-stride))))
+
+(defn vertex-buffer-to-clj
+  [^VertexBuffer vbuf]
+  (let [{:keys [attributes ^long size]} (.vertex-description vbuf)
+        vertex-count (count vbuf)
+        buf (.buf vbuf)]
+    {:bytes (* vertex-count size)
+     :count vertex-count
+     :stride size
+     :verts (buf-clj-vertex-data buf attributes size)}))
+
+(defn vertex-buffer-print-data [^VertexBuffer vbuf]
+  (-> (vertex-buffer-to-clj vbuf)
+      (update :verts (fn [verts]
+                       (conj (subvec verts 0 (min 3 (count verts)))
+                             '...)))))
```

Comment:
The vertex buffers can be really long. When printing, truncate the `:verts` vector so it only includes the data from the first three vertices.